Modules/_hashopenssl.c

```diff
@@ -1879,94 +1879,98 @@ hashlib_free(void *m)
-    hashlib_traverse,
-    hashlib_clear,
-    hashlib_free
-};
-
-PyMODINIT_FUNC
-PyInit__hashlib(void)
-{
-    PyObject *m, *openssl_md_meth_names;
-    _hashlibstate *state = NULL;
-#ifdef PY_OPENSSL_HAS_SHAKE
-    PyObject *bases;
-#endif
-
+/* Py_mod_exec functions */
+static int
+hashlib_openssl_legacy_init(PyObject *module) {
 #if (OPENSSL_VERSION_NUMBER < 0x10100000L) || defined(LIBRESSL_VERSION_NUMBER)
     /* Load all digest algorithms and initialize cpuid */
     OPENSSL_add_all_algorithms_noconf();
     ERR_load_crypto_strings();
 #endif
+    return 0;
+}
 
-    m = PyState_FindModule(&_hashlibmodule);
-    if (m != NULL) {
-        Py_INCREF(m);
-        return m;
-    }
-
-    m = PyModule_Create(&_hashlibmodule);
-    if (m == NULL)
-        return NULL;
-
-    state = get_hashlib_state(m);
+static int
+hashlib_init_evptype(PyObject *module) {
+    _hashlibstate *state = get_hashlib_state(module);
 
-    PyTypeObject *EVPtype = (PyTypeObject *)PyType_FromSpec(&EVPtype_spec);
-    if (EVPtype == NULL) {
-        Py_DECREF(m);
-        return NULL;
+    state->EVPtype = (PyTypeObject *)PyType_FromSpec(&EVPtype_spec);
+    if (state->EVPtype == NULL) {
+        return -1;
     }
-    state->EVPtype = EVPtype;
-    Py_INCREF((PyObject *)state->EVPtype);
-    PyModule_AddObject(m, "HASH", (PyObject *)state->EVPtype);
-
-    PyTypeObject *HMACtype = (PyTypeObject *)PyType_FromSpec(&HMACtype_spec);
-    if (HMACtype == NULL) {
-        Py_DECREF(m);
-        return NULL;
+    if (PyModule_AddType(module, state->EVPtype) < 0) {
+        return -1;
     }
-    state->HMACtype = HMACtype;
-    Py_INCREF((PyObject *)state->HMACtype);
-    PyModule_AddObject(m, "HMAC", (PyObject *)state->HMACtype);
+    return 0;
+}
 
+static int
+hashlib_init_evpxoftype(PyObject *module) {
 #ifdef PY_OPENSSL_HAS_SHAKE
-    bases = PyTuple_Pack(1, (PyObject *)EVPtype);
+    _hashlibstate *state = get_hashlib_state(module);
+    PyObject *bases;
+
+    if (state->EVPtype == NULL) {
+        return -1;
+    }
+
+    bases = PyTuple_Pack(1, state->EVPtype);
     if (bases == NULL) {
-        Py_DECREF(m);
-        return NULL;
+        return -1;
     }
-    PyTypeObject *EVPXOFtype = (PyTypeObject *)PyType_FromSpecWithBases(
+
+    state->EVPXOFtype = (PyTypeObject *)PyType_FromSpecWithBases(
         &EVPXOFtype_spec, bases
     );
     Py_DECREF(bases);
-    if (EVPXOFtype == NULL) {
-        Py_DECREF(m);
-        return NULL;
+    if (state->EVPXOFtype == NULL) {
+        return -1;
+    }
+    if (PyModule_AddType(module, state->EVPXOFtype) < 0) {
+        return -1;
     }
-    state->EVPXOFtype = EVPXOFtype;
-
-    Py_INCREF((PyObject *)state->EVPXOFtype);
-    PyModule_AddObject(m, "HASHXOF", (PyObject *)state->EVPXOFtype);
 #endif
+    return 0;
+}
 
-    openssl_md_meth_names = generate_hash_name_list();
-    if (openssl_md_meth_names == NULL) {
-        Py_DECREF(m);
-        return NULL;
+static int
+hashlib_init_hmactype(PyObject *module) {
```

Comment:
PEP 7:

```suggestion
static int
hashlib_init_hmactype(PyObject *module)
{
```

Same remark for the othe init functions.